bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/CreateBuilder.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *pa
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import org.apache.bookkeeper.client.BKException;
+
+/**
+ * Builder-style interface to create new ledgers
+ */
+public interface CreateBuilder extends OpBuilder<WriteHandle> {
+
+    CreateBuilder withEnsembleSize(int ackQuorumSize);
+
+    CreateBuilder withWriteQuorumSize(int ackQuorumSize);
+
+    CreateBuilder withAckQuorumSize(int ackQuorumSize);
+
+    CreateBuilder withPassword(byte[] password);
+
+    CreateBuilder withCustomMetadata(Map<String, byte[]> customMetadata);
+
+    CreateBuilder withDigestType(org.apache.bookkeeper.client.BookKeeper.DigestType digestType);
+
+    CreateAdvBuilder makeAdv();
+
+    WriteHandle create() throws BKException, InterruptedException;
```

Comment:
I saw some builder has sync method, while some aren't. can we make them consistent (e.g. removing the sync method)?

Reply:
I left sync methods only for create() and open() to make it simpler to use try-with-resources and have a simpler API for newbies

Reply:
as I suggested before in one of the comment, why can't you have a util function on retrieving a result from the completablefure?

at least, I would prefer consistency across the builders. it is going to be confusing if there are inconsistency.

Reply:
Dropped